source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,287 @@
+For information about the Arch of your Synology look at the <a href="https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model">list of architectures on spkrc</a>
+Depending on your device, compilation may take a hour or more (significantly less if you have a SSD and a good CPU).
+</p>
 
-Create homeassistant config directory & switch to it
+Compile Python3 for your Synology model
+```bash
+# make arch-XXXX
+```
+After the compilation is done, you can find the Python package at “*~/spksrc/spksrc/packages/python3_XXXX.spk*”.
+It should be named something like “python3_armada370-6.1_3.5.5-7.spk”, of course with a possibly different arch and version.
 
+Now you need to extract your “cross compiled module package” files which you added earlier.
+For ease of use this guide will use p7zip for extracting from these archives.
+#### Ubuntu/Mint
+```bash
+# sudo apt-get install p7zip
+```
+#### Solus
 ```bash
-# mkdir /volume1/homeassistant
-# chown homeassistant /volume1/homeassistant 
-# chmod 755 /volume1/homeassistant
-# cd /volume1/homeassistant
+# sudo eopkg install p7zip
 ```
-Hint: alternatively you can also create a "Shared Folder" via Synology WebUI (e.g., via "File Station") - this has the advantage that the folder is visible via "File Station".
+Run these commands to extract the packages
+```bash
+# 7z e python3_XXXX.spk package.tgz
+# 7z e package.tgz package.tar
+# 7z e package.tar -o"Module-Packages" "share/wheelhouse/cffi-1.11.5-cp35-none-any.whl"
+# 7z e package.tar -o"Module-Packages" "share/wheelhouse/bcrypt-3.1.4-cp35-none-any.whl"
+# 7z e package.tar -o"Module-Packages" "share/wheelhouse/cryptography-2.3.1-cp35-none-any.whl"
+# 7z e package.tar -o"Module-Packages" "share/wheelhouse/pycryptodome-3.7.2-cp35-none-any.whl"
+```
+This should give you a directory named “**Module-Packages**” with four .whl files, which you need to install later.
+Inside some of the .whl archives you need to rename all files containing the text “**x86_64-linux-gnu**” to “**arm-linux-gnueabihf**”.
+If you do not, these will be ignored by Python on ARM based Synology's, causing Home Assistant to not function.
+```bash
+# 7z rn "Module-Packages/cffi-1.11.5-cp35-none-any.whl" "_cffi_backend.cpython-35m-x86_64-linux-gnu.so" "_cffi_backend.cpython-35m-arm-linux-gnueabihf.so"
+# 7z l "Module-Packages/pycryptodome-3.7.2-cp35-none-any.whl" | grep "x86_64-linux-gnu" | cut -c54- | while read -r file; do 7z rn "Module-Packages/pycryptodome-3.7.2-cp35-none-any.whl" "$file" "$(echo $file | sed "s/x86_64-linux-gnu/arm-linux-gnueabihf/")"; done
+```
+<p class='note'>
+The 7z <b>rn</b> (eg, rename) parameter was included from 7-Zip 9.30, in the case you are pitifully stuck on a older version, do as follows.
+Extract the .whl (these are zip archives), rename all files as described above, then rearchive as zip (ofcourse the same full name with .whl and without .zip).
```

Comment:
```suggestion
Extract the .whl (these are zip archives), rename all files as described above, then rearchive as zip (use the same full name with `.whl` as extension and not `.zip`).
```

Reply:
Is there a reason you used backticks instead of quotes?

Reply:
Backticks allows the code to be formatted properly.

Reply:
In what cases should is using backticks appropriate?
it is not clear to me yet the use case between quotes and backticks.

Right now I understand backticks here are to show part of the filename, but should I be using backticks in all places were something might be used in commands or filenames?